### PR TITLE
refactor alert rules

### PR DESCRIPTION
**Pull request by umezawatakeshi**
Signed-off-by: UMEZAWA Takeshi <takeshi-umezawa@cybozu.co.jp>

### CODE REVIEW

File: monitoring/base/prometheus/alert_rules/metallb.yaml [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
-          severity: critical
+          severity: error
+        for: 10m
+        annotations:
+          summary: MetalLB controller has disappeared from Prometheus target discovery.
+          runbook: Please consider to find root causes, and solve the problems
+      - alert: MetalLBSpeakerDown
+        expr: |
+          up{job="metallb-speaker"}==0
+            unless on(node) kube_node_spec_taint{key="node.kubernetes.io/unreachable",effect="NoExecute"}
```
Comment (ymmt2005) [inline monitoring/base/prometheus/alert_rules/metallb.yaml:16]: DaemonSet has a toleration for `node.kubernetes.io/unreachable`.
Why don't you use CKE's taints? https://github.com/cybozu-go/cke/blob/main/docs/sabakan-integration.md

File: monitoring/base/victoriametrics/rules/metallb-alertrule.yaml [hunk 2]
```diff
@@ -7,14 +7,24 @@ spec:
...
-            severity: critical
+            severity: error
+          for: 10m
+          annotations:
+            summary: MetalLB controller has disappeared from Prometheus target discovery.
+            runbook: Please consider to find root causes, and solve the problems
+        - alert: MetalLBSpeakerDown
+          expr: |
+            up{job="metallb-speaker"}==0
+              unless on(node) kube_node_spec_taint{key="node.kubernetes.io/unreachable",effect="NoExecute"}
```
Comment (ymmt2005) [inline monitoring/base/victoriametrics/rules/metallb-alertrule.yaml:22]: ditto

File: monitoring/base/prometheus/alert_rules/network-policy.yaml [hunk 3]
```diff
@@ -3,7 +3,8 @@ groups:
     rules:
       - alert: CalicoNodeDown
         expr: |
-          absent(up{job="calico-node"} == 1)
+          up{job="calico-node"}==0
+            unless on(node) kube_node_spec_taint{key="node.kubernetes.io/unreachable",effect="NoExecute"}
```
Comment (ymmt2005) [inline monitoring/base/prometheus/alert_rules/network-policy.yaml:7]: ditto

File: monitoring/base/victoriametrics/rules/network-policy-alertrule.yaml [hunk 4]
```diff
@@ -9,7 +9,8 @@ spec:
       rules:
         - alert: CalicoNodeDown
           expr: |
-            absent(up{job="calico-node"} == 1)
+            up{job="calico-node"}==0
+              unless on(node) kube_node_spec_taint{key="node.kubernetes.io/unreachable",effect="NoExecute"}
```
Comment (ymmt2005) [inline monitoring/base/victoriametrics/rules/network-policy-alertrule.yaml:13]: ditto

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

File: monitoring/base/prometheus/alert_rules/etcd.yaml [hunk 5]
```diff
@@ -26,15 +26,33 @@ groups:
...
-        expr: absent(up{job="bootserver-etcd"} == 1)
+        # "3" is total number of etcd servers
+        expr: absent(up{job="bootserver-etcd"} == 1) or count(up{job="bootserver-etcd"} == 1) < 3
+        labels:
+          severity: warning
+        for: 20m
+        annotations:
+          summary: BootserverEtcd has disappeared from Prometheus target discovery.
+          runbook: Please consider to find root causes, and solve the problems
+      - alert: BootserverEtcdMissing
```
Comment (morimoto-cybozu) [inline monitoring/base/prometheus/alert_rules/etcd.yaml:37]: Is it common practice to set 2 alerts with the same name while differentiating their severity?

File: monitoring/base/prometheus/alert_rules/metallb.yaml (hunk 1 above)
Comment (umezawatakeshi) [reply monitoring/base/prometheus/alert_rules/metallb.yaml:16]: Unfortunately, `cke.cybozu.com/state` taint is not added during all-servers reboot due to CKE's problem. We cannot use it for unreachable detection currently.

Even if DaemonSet has a toleration for `node.kubernetes.io/unreachable`, this is not a problem. This alert expression uses the taint just for alert suppression.

Comment (ymmt2005) [reply monitoring/base/prometheus/alert_rules/metallb.yaml:16]: Alright.

**Review by ymmt2005 — DISMISSED** [review]

File: monitoring/base/prometheus/alert_rules/etcd.yaml (hunk 5 above)
Comment (umezawatakeshi) [reply monitoring/base/prometheus/alert_rules/etcd.yaml:37]: Yes. Alertmanager can inhibit alerts with lower severity. See inhibit_rules in [Alertmanager Example](https://github.com/prometheus/alertmanager/blob/master/README.md#example).

**Review by ymmt2005 — DISMISSED** [review]
LGTM

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**